lib/pure/options.nim

```diff
@@ -422,11 +422,6 @@ when isMainModule:
       check some("a") != stringNone
       check intNone == intNone
 
-      when compiles(some("a") == some(5)):
-        check false
-      when compiles(none(string) == none(int)):
-        check false
-
```

Comment:
Keep these checks they are part of the tests. Do not remove useful tests just because they happen to use ``system.compiles``.